dataland-community-manager/src/main/kotlin/org/dataland/datalandcommunitymanager/services/SingleDataRequestEmailSender.kt

```diff
@@ -0,0 +1,91 @@
+package org.dataland.datalandcommunitymanager.services
+
+import org.dataland.datalandbackend.model.enums.p2p.DataRequestCompanyIdentifierType
+import org.dataland.datalandcommunitymanager.model.dataRequest.SingleDataRequest
+import org.dataland.datalandemail.email.EmailSender
+import org.dataland.keycloakAdapter.auth.DatalandJwtAuthentication
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Service
+
+/**
+ * Implementation of a request manager service for all operations concerning the processing of single data requests
+ */
+@Service
+class SingleDataRequestEmailSender(
+    @Autowired private val emailSender: EmailSender,
+    @Autowired private val singleDataRequestEmailBuilder: SingleDataRequestEmailBuilder,
+    @Autowired private val singleDataRequestInternalEmailBuilder: SingleDataRequestInternalEmailBuilder,
+) {
+    /**
+     * Sends emails to the proper recepients,
+     * i.e. to the provided contacts and the data owners of the specified companies
+     * or the dataland staff if the company is not known to Dataland
+     * or no contact is specified and no data owner is known
+     * @param userAuthentication the authentication of the user who called this method
+     * @param singleDataRequest the fundamental data request
+     * @param companyIdentifierType the type of the identifer provided by the user or the Dataland company ID
+     * @param companyIdentifierValue an identifier for the company of type [companyIdentifierType]
+     */
+    fun sendSingleDataRequestEmails(
+        userAuthentication: DatalandJwtAuthentication,
+        singleDataRequest: SingleDataRequest,
+        companyIdentifierType: DataRequestCompanyIdentifierType,
+        companyIdentifierValue: String,
+    ) {
+        if (singleDataRequest.listOfReportingPeriods.isEmpty()) return
+        if (companyIdentifierType != DataRequestCompanyIdentifierType.DatalandCompanyId) {
+            sendInternalEmail(
```

Comment:
i just thought about the sendInternalEmail() statement.
you call this at 2 occasions in this function.
first one here and the second one if the contactList is empty.
Couldn't you put that in one if statement to avoid the duplicate call of this function?
Would be even more compressed if the sendEmailToSpecifiedContacts() can be streamlined or removed in case of the second if block (please adress the comment below first)

Reply:
I did it because that was the logical order. Send emails to contacts but if there are none still send one to dataland staff. You are right, these could be joined. I don't have a strong opinion on this. So I'll join the calls.

Reply:
What do you mean by streamlining the sendEmailToSpecifiedContacts()?

Reply:
i was referring to line 48, but nvm